src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/LongPollingTransport.java

```diff
@@ -150,23 +156,28 @@ public void setOnClose(TransportOnClosedCallback onCloseCallback) {
 
     @Override
     public Completable stop() {
-        if (!stopCalled.get()) {
-            this.stopCalled.set(true);
+        if (stopCalled.compareAndSet(false, true)) {
             this.active = false;
-            return this.updateHeaderToken().flatMapCompletable((x) -> {
+            return this.updateHeaderToken().doOnComplete(() -> {
                 HttpRequest request = new HttpRequest();
                 request.addHeaders(headers);
                 this.pollingClient.delete(this.url, request);
+            }).andThen(Completable.defer(() -> {
                 CompletableSubject stopCompletableSubject = CompletableSubject.create();
-                return this.receiveLoop.andThen(Completable.defer(() -> {
-                    logger.info("LongPolling transport stopped.");
-                    this.onReceiveThread.shutdown();
-                    this.threadPool.shutdown();
-                    this.onClose.invoke(this.closeError);
-                    return Completable.complete();
-                })).subscribeWith(stopCompletableSubject);
+                return this.receiveLoop.doOnComplete(() -> {
```

Comment:
Could we hook this up before the call to stop? That way you wouldn't need to set up an onErrorResumeNext callback to call stope().

Reply:
Looks promising, I'll give it a try

Reply:
While doing this I ended up fixing a scenario where we could leak a thread or two